src/domain/request/usecase/ExecuteReadyRequestsUseCase.js

```diff
@@ -0,0 +1,70 @@
+const Response = require('../../response/Response');
+
+class ExecuteReadyRequestsUseCase {
+  constructor(
+    fetchDataUseCase,
+    selectDataUseCase,
+    sendResponseToOracleUseCase,
+    requestRepository,
+    responseRepository,
+    logger,
+  ) {
+    this.fetchDataUseCase = fetchDataUseCase;
+    this.selectDataUseCase = selectDataUseCase;
+    this.sendResponseToOracleUseCase = sendResponseToOracleUseCase;
+    this.requestRepository = requestRepository;
+    this.responseRepository = responseRepository;
+    this.logger = logger;
+  }
+
+  async executeReadyRequests() {
+    const requests = await this.requestRepository.getReadyRequests();
+
+    const promises = requests.map(async (request) => {
+      let response;
```

Comment:
Response is assigned only once. Splitting this long function into a few smaller ones should help increase readability and replace this `let` with `const`.

Reply:
Let is needed here because of scope visibility